src/utils/xhr-loader.ts

```diff
@@ -164,37 +164,71 @@ class XhrLoader implements Loader<LoaderContext> {
         const status = xhr.status;
         // http status between 200 to 299 are all successful
         if (status >= 200 && status < 300) {
```

Comment:
Rather than wrap all the code below in a try catch and duplicate all the retry code why don't we just change the if statement above?

```suggestion
        if (
          status >= 200 &&
          status < 300 &&
          (xhr.response || xhr.responseText)
        ) {
```

This way, if `xhr.response` is null we won't enter the success block and will instead treat it like an error.